src/utilities/transform-navigation-structure.ts

```diff
@@ -0,0 +1,41 @@
+const navigationStructure = {
+  overview: [],
+  theme: [],
+  components: {}
+}
+const nestedNavigationCategories = {
+  components: ['Overview', 'General', 'Layout', 'Media', 'Forms', 'Utilities']
+}
+
+const sortWithinSource = (categories) => (a, b) =>
+  // sort by category name
+  categories.indexOf(a.category) - categories.indexOf(b.category) ||
+  // sort by priority within that category, default to compare against high number
```

Comment:
again just for my comprehension, where does the priority get set?

Reply:
Priority gets set in the frontmatter of the markdown docs, e.g.

```md
---
title: Yolo
category: Hello world
priority: 1
---
```

Reply:
oh it is being introduced now then. Thanks for explaining.